server/src/main/java/io/crate/auth/JWTAuthenticationMethod.java

```diff
@@ -67,16 +70,18 @@ public class JWTAuthenticationMethod implements AuthenticationMethod {
 
     public JWTAuthenticationMethod(Roles roles,
                                    Settings settings,
-                                   Function<String, JwkProvider> urlToJwkProvider,
+                                   Map<String, JwkProvider> cachedJwkProviders,
+                                   BiFunction<String, Duration, JwkProvider> jwkProvider,
                                    Supplier<String> clusterId) {
         this.roles = roles;
         this.settings = settings;
-        this.urlToJwkProvider = urlToJwkProvider;
+        this.jwkProvider = jwkProvider;
+        this.cachedJwkProviders = cachedJwkProviders;
         this.clusterId = clusterId;
     }
 
     @Override
-    public @Nullable Role authenticate(Credentials credentials, ConnectionProperties connProperties) {
+    public @Nullable Role authenticate(Credentials credentials, @Nullable ConnectionProperties connProperties) {
```

Comment:
ConnectionProperties cannot be nullable?

Reply:
That's a leftover. It used to be null in the test, but this can be removed now.